src/core/src/bootstrap/CoreUtility.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+import re
+
+
+class CoreUtility(object):
+
+    @staticmethod
+    def compare_version(version_a, version_b):
+        """ Compare two versions with handling numeric and string parts, return -1 (less), +1 (greater), 0 (equal) """
+
+        def split_version_components(version):
+            """ Split a version into numeric and non-numeric into components list: 27.13.4~18.04.1 -> [27][14][4]"""
```

Comment:
Please include # type for all methods as previously discussed.

Reply:
done